### PR TITLE
rvc add r_vc_find_rp ##projects

### PR DESCRIPTION
**Pull request by RHL120**
- [ ] Closing issues: #issue
- [X] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

### CODE REVIEW

File: libr/core/rvc.c [hunk 1]
```diff
@@ -654,6 +654,51 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+R_API char *r_vc_find_rp(const char *path) {
+	{
+		int ret = repo_exists (path);
+		switch (ret) {
+		case 0:
+			break;
+		case 1:
+			return r_str_new (path);
+		case -1:
+			printf ("A corrupted repo may have been found at %s, please refrain from naming any files .rvc\n", path);
```
Comment (trufae) [inline libr/core/rvc.c:666]: Eprintf and missing break

File: libr/core/rvc.c [hunk 2]
```diff
@@ -654,6 +654,51 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+		}
+	}
+	const char *p = r_str_rchr (path, path + strlen (path), *R_SYS_DIR);
+	if (!p) {
+		return NULL;
+	}
+	char *i = malloc (p - path + 1);
+	if (!i) {
+		return NULL;
+	}
```
Comment (trufae) [inline libr/core/rvc.c:676]: Use r_str_ndup

File: libr/core/rvc.c [hunk 3]
```diff
@@ -654,6 +654,51 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+		case 1:
+			return i;
+		case -1:
+			printf ("A corrupted repo may have been found at %s, please refrain from naming any files .rvc\n", i);
+		}
+		p = r_str_rchr (path, p, *R_SYS_DIR);
+		if (!p) {
+			return NULL;
+		}
+		i = malloc (p - path + 1);
```
Comment (trufae) [inline libr/core/rvc.c:693]: Str_ndup

File: libr/core/rvc.c [hunk 4]
```diff
@@ -654,6 +654,51 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+	printf ("i without %s\n", i); //dbgtrm
+	while (true) {
+		int ret = repo_exists (i);
+		switch (ret) {
+		case 0:
+			break;
+		case 1:
+			return i;
+		case -1:
+			printf ("A corrupted repo may have been found at %s, please refrain from naming any files .rvc\n", i);
```
Comment (trufae) [inline libr/core/rvc.c:687]: Eprintf and break

File: libr/core/rvc.c [hunk 5]
```diff
@@ -654,6 +654,51 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+	const char *p = r_str_rchr (path, path + strlen (path), *R_SYS_DIR);
+	if (!p) {
+		return NULL;
+	}
+	char *i = malloc (p - path + 1);
+	if (!i) {
+		return NULL;
+	}
+	strncpy (i, path, p - path);
+	printf ("i without %s\n", i); //dbgtrm
```
Comment (trufae) [inline libr/core/rvc.c:678]: Eprintf or rm completely. Apis shouldnt print stuff to console

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/rvc.c (hunk 5 above)
Comment (RHL120) [reply libr/core/rvc.c:678]: Yes, dbgtrm stands for debug to remove I forgot to run g/\/\/dbgtrm/norm!dd

File: libr/core/rvc.c [hunk 6]
```diff
@@ -654,6 +654,49 @@ static RList *blobs_add(const char *rp, const RList *files) {
...
+R_API char *r_vc_find_rp(const char *path) {
+	{
+		int ret = repo_exists (path);
+		switch (ret) {
+		case 0:
+			break;
+		case 1:
+			return r_str_new (path);
+		case -1:
+			printf ("A corrupted repo may have been found at %s, please refrain from naming any files .rvc\n", path);
```
Comment (trufae) [inline libr/core/rvc.c:666]: Its not resolved . Use eprintf and add a break

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/rvc.c (hunk 6 above)
Comment (RHL120) [reply libr/core/rvc.c]: Sorry about that mixed it up with the one above

**Pull request merged**